contracts/MechMarketplace.sol

```diff
@@ -199,103 +246,46 @@ contract MechMarketplace {
-        emit OwnerUpdated(newOwner);
-    }
-
-    /// @dev Changes mech factory address.
-    /// @param newFactory New mech factory address.
-    function changeFactory(address newFactory) external {
-        // Check for the contract ownership
-        if (msg.sender != owner) {
-            revert OwnerOnly(msg.sender, owner);
-        }
-
-        // Check for zero address
-        if (newFactory == address(0)) {
-            revert ZeroAddress();
-        }
-
-        factory = newFactory;
-        emit FactoryUpdated(newFactory);
-    }
-
-    /// @dev Changes min and max response timeout values.
-    /// @param newMinResponseTimeout New min response timeout.
-    /// @param newMaxResponseTimeout New max response timeout.
-    function changeMinMaxResponseTimeout(uint256 newMinResponseTimeout, uint256 newMaxResponseTimeout) external {
-        // Check contract ownership
-        if (msg.sender != owner) {
-            revert OwnerOnly(msg.sender, owner);
-        }
-
-        // Check for zero values
-        if (newMinResponseTimeout == 0 || newMaxResponseTimeout == 0) {
-            revert ZeroValue();
-        }
-
-        // Check for sanity values
-        if (newMinResponseTimeout > newMaxResponseTimeout) {
-            revert Overflow(newMinResponseTimeout, newMaxResponseTimeout);
-        }
-
-        // responseTimeout limits
-        if (newMaxResponseTimeout > type(uint32).max) {
-            revert Overflow(newMaxResponseTimeout, type(uint32).max);
-        }
-
-        minResponseTimeout = newMinResponseTimeout;
-        maxResponseTimeout = newMaxResponseTimeout;
-        
-        emit MinMaxResponseTimeoutUpdated(newMinResponseTimeout, newMaxResponseTimeout);
-    }
-
-    /// @dev Sets mech registration status.
-    /// @param mech Mech address.
-    /// @param status True, if registered, false otherwise.
-    function setMechRegistrationStatus(address mech, bool status) external {
-        // Check for the agent mech factory access, contract ownership or mech itself (when changing its marketplace)
-        if (msg.sender != factory && msg.sender != owner && msg.sender != mech) {
-            revert OwnerOnly(msg.sender, owner);
+    /// @dev Checks agent mech for contract validity.
+    /// @param mech Agent mech address.
+    /// @param mechServiceId Mech operator service Id.
+    function checkMech(address mech, uint256 mechServiceId) public view {
+        // Check that the mech address corresponds to the authorized bytecode hash
+        bytes32 mechHash = keccak256(mech.code);
+        if (mechHash != mechBytecodeHash) {
+            revert UnauthorizedAccount(mech);
         }
 
-        // Prevent mech calling this function by exec() and trying to whitelist itself
-        if (msg.sender == mech && status) {
-            revert UnauthorizedAccount(msg.sender);
+        // Check if the mech service is staked
+        IStaking.StakingState state = IStaking(mechStakingInstance).getStakingState(mechServiceId);
```

Comment:
why are we not making mechStakingInstance an argument in this method? It seems this is now scoped to a single hardcoded mechStakingInstance.

Reply:
Not relevant anymore as part of the ongoing PR.